app/views/play/level/tome/SpellView.coffee

```diff
@@ -694,6 +705,41 @@ module.exports = class SpellView extends CocoView
     return if @destroyed
     Backbone.Mediator.publish 'tome:hide-problem-alert', {}
 
+  saveSpade: (e) =>
+    return if @destroyed
+    spadeEvents = @spade.compile()
+    # Uncomment the below line for a debug panel to display inside the level
+    #@spade.debugPlay(spadeEvents)
+    condensedEvents = @spade.condense(spadeEvents)
+    return unless condensedEvents.length
+    compressedEvents = LZString.compressToUTF16(JSON.stringify(condensedEvents))
+    codelog = new CodeLog({
+      sessionID: @options.session.id
+      level:
+        original: @options.level.get 'original'
+        majorVersion: (@options.level.get 'version').major
+      levelSlug: @options.level.get 'slug'
+      userID: @options.session.get 'creator'
+      userName: @options.session.get 'creatorName'
+      log: compressedEvents
+    })
+    codelog.on('save:success', (e) ->
```

Comment:
`e` should be `codelog`, since that's what the first argument is. `e` is for "event", for jquery event callbacks and such.